### DISCUSSION TITLE
Library wrapping question, simple vs redirected

### DISCUSSION THREAD

**Discussion by Javier97sm** [ptitSeb/box64#1827] [Q&A]
Hello, I am trying to wrap a new library (libjack) so I've been reading the following guide (https://box86.org/2024/02/how-to-create-a-wrapping/) and also taking a look at other wrappings like pulseaudio. 

I don't understand when I have to add simple functions with the GO() macro or use GOM(). I see it clear with variadic functions but not very clear for redirected functions. For example, this pulseaudio functions:

GO(pa_threaded_mainloop_accept, vFp)
GOM(pa_threaded_mainloop_free, vFEp)

When you take a look at the pulseaudio source code, this is the functions signature:

void pa_threaded_mainloop_accept(pa_threaded_mainloop *m);
void pa_threaded_mainloop_free(pa_threaded_mainloop* m);

Both have the same signature, where its argument 'pa_threaded_mainloop' is a pointer to a pretty big struct. But one has its simple and the other redirected. Can someone help me understand the reasoning behind this? 

Thanks in advance!

**Comment by ptitSeb** [comment]
For this particular example, it's because on `pa_thread_mainloop_free`, box64 needs to do some cleanup at this function, so it has been made `GOM` while technicaly `GO` would be enough.

I let @rajdakin also comment there.
